.github/workflows/deploy.yml

```diff
@@ -0,0 +1,38 @@
+name: Publish Image
+
+on:
+  pull_request:
+    branches:
+      - main
+  release:
+    types: [published]
+
+env:
+  REGISTRY: ghcr.io
+  IMAGE_NAME: ${{ github.repository }}
```

Comment:
Don't think IMAGE_NAME is needed here as it's set in the makefile.. That example repository is just a sample for adaption and doesn't need to be copied verbatim.